File: python/day01b.py
from collections import deque


def main():
    with open("input/day01.txt") as file:
        result = 0
        window = deque()
        for line in file.readlines():
            window.append(int(line))
            if len(window) == 4:
                if window[-1] > window[0]:
                    result += 1
                window.popleft()
        print(f"{result} sums are larger")


if __name__ == "__main__":
    main()

File: src/bin/day01b.rs
use std::collections::VecDeque;
use std::fs::File;
use std::io::{BufRead, BufReader};

type Error = Box<dyn std::error::Error>;

fn main() -> Result<(), Error> {
    let file = File::open("input/day01.txt")?;
    let reader = BufReader::new(file);
    let mut window = VecDeque::new();
    let mut result = 0;
    for line in reader.lines() {
        window.push_back(line?.parse::<i32>()?);
        if window.len() == 4 {
            if window.back() > window.front() {
                result += 1;
            }
            window.pop_front();
        }
    }
    println!("{} sums are larger", result);
    Ok(())
}
